lms/resources/lti_launch.py

```diff
@@ -49,21 +59,12 @@ def h_authority_provided_id(self):
             self._request.parsed_params["tool_consumer_instance_guid"].encode()
         )
         hash_object.update(self._request.parsed_params["context_id"].encode())
-        return hash_object.hexdigest()
-
-    @property
-    def h_groupid(self):
-        """
-        Return a unique h groupid for the current request.
-
-        The returned ID is suitable for use with the h API's ``groupid`` parameter.
+        authority_provided_id = hash_object.hexdigest()
 
-        The groupid is deterministic and is unique to the LTI course. Calling this
-        function again with params representing the same LTI course will always
-        return the same groupid. Calling this function with different params will
-        always return a different groupid.
-        """
-        return f"group:{self.h_authority_provided_id}@{self._authority}"
+        return HGroup(
+            name=self._group_name(self._request.parsed_params["context_title"].strip()),
+            authority_provided_id=authority_provided_id,
+        )
```

Comment:
I can't really see how this is all going to come together with the various methods below. This also seems like a different pattern for creating `HGroups`, from both `LTIUsers` and `HUsers`.

I'm going to assume this is half way through a chain of stuff and doesn't make much sense now.

Reply:
Haven't really considered it yet at all